lib/thredded/truthy_hash.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+module Thredded
+  module TruthyHash
```

Comment:
went for your style in the end (though I've called it `PerNotifierPrefs`). Not validating keys (other than in strong params) and haven't (yet -- probably a future PR). allowed configuration of defaults, though this is actually relatively straightforward.